stellargraph/core/graph.py

```diff
@@ -279,6 +274,25 @@ def __init__(
             weight_column=edge_weight_column,
         )
 
+        edge_node_set = set(self._edges.sources)
+        edge_node_set.update(self._edges.targets)
+
+        if nodes is None and edges is not None:
+            nodes = pd.DataFrame([], index=edge_node_set)
+        elif nodes is None:
+            nodes = {}
+
+        self._nodes = convert.convert_nodes(
+            nodes, name="nodes", default_type=node_type_default, dtype=dtype,
+        )
+
+        node_set = set(self._nodes.ids.pandas_index)
+        if not edge_node_set.issubset(node_set):
```

Comment:
Like the `set` work above, I _suspect_ this may be better staying in the Pandas world as much as possible. In particular, we can leverage `to_iloc`'s `strict` parameter:

https://github.com/stellargraph/stellargraph/blob/5f59b166532242b05ca71038e9b8dc86ee35e177/stellargraph/core/element_data.py#L80-L102

Maybe something like:

```python
try:
    self._nodes.ids.to_iloc(edge_node_set, smaller_type=False, strict=True)
except KeyError as e:
    missing_values = e.args[0]
    if not is_real_iterable(missing_values):
        missing_values = [missing_values]

    raise ValueError(...) from None
```

Or, if we move away from `edge_node_set`, maybe:

```python
try:
    self._nodes.ids.to_iloc(pd.concat([self._edges.source, self._edges.target]), smaller_type=False, strict=True)
except KeyError as e:
    missing_values = e.args[0]
    if not is_real_iterable(missing_values):
        missing_values = [missing_values]
    missing_values = pd.unique(missing_values)

    raise ValueError(...) from None
```

Reply:
I tried this approach and it works much better!

![image](https://user-images.githubusercontent.com/23410124/76589347-e6559400-653d-11ea-83a9-70739478d700.png)